src/main/java/seedu/address/model/person/ReminderManager.java

```diff
@@ -3,92 +3,175 @@
         }
 
-        return reminders;
+        // Add "and X more" if there are additional people
+        if (eligiblePeople.size() > MAX_NAMES_TO_SHOW) {
+            names.append(String.format(MORE_SUFFIX, eligiblePeople.size() - MAX_NAMES_TO_SHOW));
+        }
+
+        return names.toString();
     }
 
-    /**
-     * Fetches the next reminder in the rotation.
-     * If multiple persons share the same deadline, this will rotate through them.
-     * @return The next reminder message.
-     */
-    public String getNextReminder() {
-        List<String> reminders = getLatestReminders();
-        if (reminders.isEmpty()) {
-            currentReminderIndex = 0;
-            return MESSAGE_NO_REMINDERS;
-        } else {
-            if (currentReminderIndex >= reminders.size()) {
-                currentReminderIndex = 0;
+    private List<String> getLatestReminders() {
+        LocalDate now = LocalDate.now();
+        List<String> reminders = new ArrayList<>();
+
+        // Collect all overdue persons (only active clients)
+        List<Person> overduePersons = new ArrayList<>();
+        for (Person person : persons) {
+            if (!person.getProjectStatus().isComplete
+                    && person.getClientStatus().toString().equals(ACTIVE_STATUS)
+                    && person.getDeadline().isOverdue()) {
+                overduePersons.add(person);
             }
+        }
 
-            String nextReminder = reminders.get(currentReminderIndex);
-            currentReminderIndex++;
+        // If there are overdue projects, show them first
+        if (!overduePersons.isEmpty()) {
+            String names = formatNames(overduePersons);
+            if (!names.isEmpty()) {
+                reminders.add(names
+                        + (overduePersons.size() == 1 ? MESSAGE_SINGLE_DEADLINE_OVERDUE
+                        : MESSAGE_DEADLINE_OVERDUE)
+                        + PERIOD);
+                return reminders;
+            }
+        }
 
-            if (currentReminderIndex >= reminders.size()) {
-                currentReminderIndex = 0;
+        // Handle upcoming deadlines if no overdue ones
+        Map<LocalDate, List<Person>> deadlineGroups = new LinkedHashMap<>();
+        for (Person person : persons) {
+            if (!person.getProjectStatus().isComplete
+                    && person.getClientStatus().toString().equals(ACTIVE_STATUS)) {
+                Deadline deadline = person.getDeadline();
+                if (!deadline.isOverdue()) {
+                    deadlineGroups.computeIfAbsent(deadline.value, k -> new ArrayList<>()).add(person);
+                }
             }
+        }
 
-            return nextReminder;
+        if (!deadlineGroups.isEmpty()) {
+            LocalDate nearestDate = deadlineGroups.keySet().stream()
+                    .min(LocalDate::compareTo)
+                    .orElse(null);
+
+            if (nearestDate != null) {
```

Comment:
Small nit: By any chance, can this function be done in a way so it looks less like an arrow shape?

Reply:
Thanks for spotting that! I've flattened out the nested structure to make it easier to read. Does it look better now?

Reply:
Thanks for spotting that! I've flattened out the nested structure to make it easier to read. Does it look better now?